homepage/models.py

```diff
@@ -12,26 +13,84 @@ class Course(models.Model):
-    syllabi = models.CharField(max_length=200)  # link to syllabi of the course, not for use, currently
+    syllabi = models.URLField(blank=True, null=True)  # link to syllabi of the course, not for use, currently
     avg_load = models.DecimalField(max_digits=6, decimal_places=5,
-                                   validators=[MinValueValidator(1), MaxValueValidator(5)])  # average course load
+                                   validators=[MinValueValidator(1), MaxValueValidator(5)],
+                                   blank=True, null=True)  # average course load, starts as null
     avg_rating = models.DecimalField(max_digits=6, decimal_places=5,
-                                     validators=[MinValueValidator(1), MaxValueValidator(5)])  # average course rating
-    num_of_raters = models.IntegerField(
-        validators=[MinValueValidator(0)])  # number of raters for course rating and course load
-    num_of_reviewers = models.IntegerField(
-        validators=[MinValueValidator(0)])  # number of reviewers
+                                     validators=[MinValueValidator(1), MaxValueValidator(5)],
+                                     blank=True, null=True)  # average course rating, starts as null
+    num_of_raters = models.IntegerField(validators=[MinValueValidator(0)],
+                                        default=0)  # number of raters for course rating and course load
+    num_of_reviewers = models.IntegerField(validators=[MinValueValidator(0)],
+                                           default=0)  # number of reviewers
 
     def __str__(self):
         return self.name
 
+    def clean(self):
+        rating = self.avg_rating
+        load = self.avg_load
+        raters = self.num_of_raters
+        reviews = self.num_of_reviewers
+
+        if (rating is not None) and (load is None):
+            raise ValidationError(("Can't have rating without load"))
+        if (rating is None) and (load is not None):
+            raise ValidationError(("Can't have load without rating"))
+        if ((rating is None) or (load is None)) and (raters > 0):
+            raise ValidationError(("Can't have raters that didn't rate"))
+        if ((rating is not None) or (load is not None)) and (raters == 0):
+            raise ValidationError(("Can't have ratings without raters"))
+        if ((raters == 0) or (load is None) or (rating is None)) and (reviews > 0):
+            raise ValidationError(("Can not have reviewes without ratings"))
+        if (reviews > raters):
+            raise ValidationError(("Can not have reviewes without ratings"))
+
+    def save(self, *args, **kwargs):
+        try:
+            self.clean()
+        except ValidationError:
+            return
+
+        super().save(*args, **kwargs)
+
+    # updates the course according to review output
+    def upd_course_per_review(self, rating, load, has_content):

```

Comment:
```suggestion
    def update_course_per_review(self, new_rating, new_load, has_content):

```

Reply:
I think this is important -
when i first read the method i was confused and i think new_rating and new_load will increase the readability

Reply:
changed but named them reviewer_rating, reviewer_load